src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1056,7 +1056,7 @@ internal PathManager CreatePathManager(IStartConfiguration config)
         internal PreferenceSettings CreatePreferences(IStartConfiguration config)
         {
             PreferenceSettings preferences = null;
-            if (!config.StartInTestMode)
+            if (config.Preferences != null)
```

Comment:
in the D4R case, this will be false and we will go to the else block for loading the preferences?

Reply:
yes, at that point we already have a PathManager pointing to the proper paths so we will create the preferences from the expected file.

Reply:
Dont we want to do the same in the StartInTestMode also? What happens in test mode then?

Reply:
For test mode, I think it depends on if Preferences was passed through start config? Can you confirm that from unit tests?

Reply:
@reddyashish @QilongTang This is transparent for the TestMode. Yes, each test is responsible to pass its preferences or not.

Reply:
This seems like a foundational problem with this set of changes - now the tests are not testing what users are experiencing and test authors ( at least me) have no idea how the different test fixtures are interacting with preferences.

Reply:
It looks like we can just revert back to the previous code at https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Models/DynamoModel.cs#L681 to simply be `PreferenceSettings = (PreferenceSettings)CreateOrLoadPreferences(config.Preferences);`

I guess to avoid confusion from @mjkkirschner , let's document what is is the difference before and after PreferenceSetting singleton change

Reply:
> It looks like we can just revert back to the previous code at https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Models/DynamoModel.cs#L681 to simply be `PreferenceSettings = (PreferenceSettings)CreateOrLoadPreferences(config.Preferences);`
> 
> I guess to avoid confusion from @mjkkirschner , let's document what is is the difference before and after PreferenceSetting singleton change

Done.